### PR TITLE
Fix bad lint fix.

### PR DESCRIPTION
**Pull request by Hoverbear**
Fixes https://github.com/pingcap/tikv/pull/2617#discussion_r161677251

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/coprocessor/codec/mysql/decimal.rs [hunk 1]
```diff
@@ -1759,7 +1759,7 @@ macro_rules! read_word {
                 }
             }
             4 => {
-                ((i32::from(u8::from(buf[0])) << 24) + (i32::from(buf[1]) << 16) +
+                ((i32::from(i8::from(buf[0])) << 24) + (i32::from(buf[1]) << 16) +
```
Comment (BusyJay) [inline src/coprocessor/codec/mysql/decimal.rs]: I think what you need is `buf[0] as i8`.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**